src/CowWrapper.sol

```diff
@@ -0,0 +1,207 @@
+
+    /// @notice Returns the authentication contract used by the settlement contract.
+    function authenticator() external view returns (ICowAuthentication);
+
+    /// @notice Returns the address of the vaultRelayer, the target for approvals for funds entering the settlement contract.
+    function vaultRelayer() external view returns (address);
+
+    /// @notice Returns the domain separator for EIP-712 signing
+    function domainSeparator() external view returns (bytes32);
+
+    /// @notice Allows for approval of orders by submitting an authorized hash on-chain prior to order execution.
+    function setPreSignature(bytes calldata orderUid, bool signed) external;
+
+    /// @notice Settles a batch of trades atomically
+    /// @param tokens Array of token addresses involved in the settlement
+    /// @param clearingPrices Array of clearing prices for each token
+    /// @param trades Array of trades to execute
+    /// @param interactions Array of three interaction arrays (pre, intra, post-settlement)
+    function settle(
+        address[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        Trade[] calldata trades,
+        Interaction[][3] calldata interactions
+    ) external;
+}
+
+/// @title CoW Protocol Wrapper Interface
+/// @notice Interface for wrapper contracts that add custom logic around CoW settlements
+/// @dev Wrappers can be chained together to compose multiple settlement operations
+interface ICowWrapper {
+    /// @notice A human readable label for this wrapper. Used for display in explorer/analysis UIs
+    function name() external view returns (string memory);
+
+    /// @notice The settlement contract used by this wrapper
+    /// @return The CowSettlement contract address
+    function SETTLEMENT() external view returns (ICowSettlement);
+
+    /// @notice Initiates a wrapped settlement call
+    /// @dev This is the entry point for wrapped settlements. The wrapper will execute custom logic
+    ///      before calling the next wrapper or settlement contract in the chain.
+    /// @param settleData ABI-encoded call to ICowSettlement.settle() containing trade data
+    /// @param wrapperData Encoded data for this wrapper and the chain of next wrappers/settlement.
+    ///                    Format: [2-byte len][wrapper-specific-data][next-address]([2-byte len][wrapper-specific-data][next-address]...)
+    function wrappedSettle(bytes calldata settleData, bytes calldata wrapperData) external;
+
+    /// @notice Parses and validates wrapper-specific data
+    /// @dev Used by CowWrapperHelpers to validate wrapper data before execution.
+    ///      Implementations should consume their portion of wrapperData and return the rest.
+    /// @param wrapperData The wrapper-specific data to parse
+    /// @return remainingWrapperData Any wrapper data that was not consumed by this wrapper
+    function parseWrapperData(bytes calldata wrapperData) external view returns (bytes calldata remainingWrapperData);
```

Comment:
The function doesn't need to return anything, `remainingWrapperData` is only intended to be used by the helper to make sure there's nothing left, but this isn't really necessary and we can just have `parseWrapperData` revert if more data is used than it should. Even if there's leftover data, this shouldn't really impact the security of the contract so it's ok to even skip it.

```suggestion
    function validateWrapperData(bytes calldata wrapperData) external view;
```

Relevant discussion is [here](https://github.com/cowprotocol/euler-integration-contracts/pull/13#discussion_r2582403270).  This function is confusing at a glance so we should make it simpler.